### PR TITLE
Adds (nonspatial) outlier functions

### PR DESCRIPTION
**Pull request by ohasselblad**
- [x] All declared geometries are `geometry(Geometry, 4326)` for general geoms, or `geometry(Point, 4326)`
- [x] Existing functions in crankshaft python library called from the extension are kept at least from version N to version N+1 (to avoid breakage during upgrades).
- [x] Docs for public-facing functions are written
- [x] New functions follow the naming conventions: `CDB_NameOfFunction`. Where internal functions begin with an underscore
- [ ] Video explaining the analysis and showing examples 
- [ ] Analysis Documentation written [template](https://docs.google.com/a/cartodb.com/document/d/1X2KOtaiEBKWNMp8UjwcLB-kE9aIOw09aOjX3oaCjeME/edit?usp=sharing)
- [ ] Smoke test written

### CODE REVIEW

**Comment by ohasselblad** [comment]
@AbelVM, could you give this one a CR, please? 

The `CDB_PercentOutlier` is a bit funny still. It accepts a ratio for the threshold but is called `Percent` so could be misleading as `2.0` is 100% larger than the average... a little counterintuitive. I also did not give the `is_symmetric` option like I did for the `CDB_StdDevOutlier` one.

File: doc/18_outliers.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+### CDB_StdDevOutlier(column_values numeric[], ratio_threshold numeric, ids int[], is_symmetric boolean DEFAULT true)
+
+`CDB_StdDevOutlier` calculates whether or not a value falls above or below a given threshold based on the number of standard deviations from the mean.
+
+#### Arguments
+
+| Name | Type | Description |
+|------|------|-------------|
+| column_values | numeric[] | An array of the values to calculate the outlier classification on |
+| num_deviations | numeric | The threshold in units of standard deviation |
```
Comment (AbelVM) [inline doc/18_outliers.md]: This argument does not have the same name as in the signature: `ratio_threshold`

File: doc/18_outliers.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+### CDB_PercentOutlier(column_values numeric[], ratio_threshold numeric, ids int[])
+
+`CDB_PercentOutlier` calculates whether or not a value falls above a given threshold based on a percentage above the mean value of the input values.
+
+#### Arguments
+
+| Name | Type | Description |
+|------|------|-------------|
+| column_values | numeric[] | An array of the values to calculate the outlier classification on |
+| outlier_fraction | numeric | The threshold above which a column value divided by the mean of all values is considered an outlier |
```
Comment (AbelVM) [inline doc/18_outliers.md]: This argument does not have the same name as in the signature: `ratio_threshold`

Comment (ohasselblad) [inline doc/18_outliers.md]: Updated in SQL signature

File: doc/18_outliers.md (hunk 1 above)
Comment (ohasselblad) [inline doc/18_outliers.md]: Updated in SQL signature

**Comment by ohasselblad** [comment]
@CartoDB/dataservices this PR is ready to be merged

**Pull request merged**